src/lib_ccx/matroska.c

```diff
@@ -738,14 +738,35 @@ void parse_segment(struct matroska_ctx* mkv_ctx)
     }
 }
 
+char *filename_without_ext(char* filename) {
+    char *returnname;
+    char *lastdot;
+    if (filename == NULL)
+         return filename;
+    if ((returnname = malloc (strlen (filename) + 1)) == NULL)
```

Comment:
If we want X chars, we should alloc (sizeof(char)*X), not just (X). This is good practice, because sizeof(char) not always equals to 1